File: scala/etl/src/main/scala/Etl.scala
object Etl {
  def transform(scores: Map[Int, Seq[String]]): Map[String, Int] =
    scores
      .flatMap { case (score, letters) => letters.map((_, score)) }
      .mapKeys(_.toLowerCase)

  // mapKeys should just be in the standard library IMO
  implicit class MapFunctions[A, B](val map: Map[A, B]) extends AnyVal {
    def mapKeys[A1](f: A => A1): Map[A1, B] = map.map({ case (a, b) => (f(a), b) })
  }
}

File: fsharp/etl/Etl.fs
module Etl

let transform (scoresWithLetters: Map<int, char list>): Map<char, int> =
    let transpose (score, letters) =
        Seq.map (fun letter -> letter, score) letters

    let lowercase (letter, score) = System.Char.ToLower letter, score

    scoresWithLetters
    |> Map.toSeq
    |> Seq.collect transpose
    |> Seq.map lowercase
    |> Map.ofSeq